File: src/main/java/org/dxworks/githubminer/dto/export/RepoDTO.java
package org.dxworks.githubminer.dto.export;

import com.google.api.client.util.Key;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.dxworks.githubminer.dto.response.repository.Repository;


@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class RepoDTO {
	@Key
	private Long id;
	@Key
	private String name;
	@Key
	private String fullName;
	@Key
	private UserDTO owner;

	public static RepoDTO fromRepository(Repository repo) {
		if (repo == null || com.google.api.client.util.Data.isNull(repo))
			return null;
		return builder()
				.id(repo.getId())
				.name(repo.getName())
				.fullName(repo.getFullName())
				.owner(UserDTO.fromUser(repo.getOwner()))
				.build();
	}
}

File: src/main/kotlin/org/dxworks/githubminer/dto/export/RepoDTO.kt
package org.dxworks.githubminer.dto.export

import com.google.api.client.util.Data
import com.google.api.client.util.Key
import org.dxworks.githubminer.dto.response.repository.Repository

class RepoDTO {
    @Key
    var id: Long? = null

    @Key
    var name: String? = null

    @Key
    var fullName: String? = null

    @Key
    var owner: UserDTO? = null

    companion object {
        fun fromRepository(repo: Repository?): RepoDTO? {
            return if (repo == null || Data.isNull(repo)) null else RepoDTO().apply {
                    id = repo.id
                    name = repo.name
                    fullName = repo.fullName
                    owner = UserDTO.fromUser(repo.owner)
            }
        }
    }
}
